.github/workflows/buildAndTest.yml

```diff
@@ -266,21 +266,19 @@ jobs:
       # Run clang-tidy against only the changes. The 'clang-tidy-diff' script
       # does this if supplied with the diff.
       - name: clang-tidy
-        if: ${{ always() }}
         run: |
-          git diff -U0 $DIFF_COMMIT...HEAD | \
-            clang-tidy-diff -path build -p1 -fix -j$(nproc)
-          git clang-format -f $DIFF_COMMIT
-          git diff --ignore-submodules > clang-tidy.patch
-          if [ -s clang-tidy.patch ]; then
-            echo "Clang-tidy problems in the following files. " \
-              "See diff in the clang-tidy.patch artifact."
-            git diff --ignore-submodules --name-only
-            git checkout .
+          FAILED=0
+          git diff -U0 $DIFF_COMMIT..HEAD | clang-tidy-diff -path build -p1 -fix -j$(nproc) || FAILED=1
+          git clang-format -f $DIFF_COMMIT || FAILED=1
```

Comment:
Does it actually make sense to run clang-format here? It seems clang-format already runs as part of Sanity Check, so I think this is just wasted effort. Thoughts on removing?

The clang-format job is here:
https://github.com/llvm/circt/blob/main/.github/workflows/buildAndTest.yml#L71

and an example run:
https://github.com/llvm/circt/actions/runs/10010063935/job/27670384820#step:7:1

Reply:
I remember clang-tidy fix is often formatted by clang-format. I think it's fine to remove though.